service/src/main/kotlin/xyz/lebkuchenfm/api/eventstream/WebSocketEventStream.kt

```diff
@@ -0,0 +1,64 @@
+package xyz.lebkuchenfm.api.eventstream
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.ktor.serialization.suitableCharset
+import io.ktor.server.routing.Route
+import io.ktor.server.websocket.converter
+import io.ktor.server.websocket.webSocket
+import io.ktor.util.reflect.typeInfo
+import io.ktor.websocket.DefaultWebSocketSession
+import io.ktor.websocket.Frame
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.ClassDiscriminatorMode
+import kotlinx.serialization.json.Json
+import xyz.lebkuchenfm.domain.eventstream.Event
+import xyz.lebkuchenfm.domain.eventstream.EventStream
+import java.util.Collections
+
+class Connection(val session: DefaultWebSocketSession)
+
+fun Route.eventStreamRouting(eventStream: WebSocketEventStream) {
+    webSocket("/event-stream") {
+        val connection = Connection(this)
+        eventStream.addConnection(connection)
+
+        for (frame in incoming) {
+            // TODO: This does not work yet, but I fix it when there are actual events coming from the client.
```

Comment:
Should we note it in the Doc?

Reply:
No, that's what's necessary for WebSockets to function properly, and that's what I'm working on now.